src/wp-includes/cron.php

```diff
@@ -417,11 +417,16 @@ function wp_reschedule_event( $timestamp, $recurrence, $hook, $args = array(), $
 	}
 
 	// Now we assume something is wrong and fail to schedule.
-	if ( 0 === $interval ) {
+	if ( ! is_numeric( $interval ) || $interval <= 0 ) {
 		if ( $wp_error ) {
 			return new WP_Error(
 				'invalid_schedule',
-				__( 'Event schedule does not exist.' )
+				sprintf(
+					/* translators: 1: the event schedule recurrence, 2: the interval encoded as JSON */
+					__( 'Event schedule with recurrence "%$1s" does not exist or has is invalid. Interval must be positive integer, but got: %$2s' ),
+					$recurrence,
+					wp_json_encode( $interval )
+				)
```

Comment:
It took me a couple of reads to understand what this is trying to tell me. The interval is an internal implementation detail that's not used by anyone calling this function, so I'm not sure it's an actionable message.

Is it possible to reliably distinguish between a schedule that doesn't exist and a schedule that exists but has an invalid interval? If so, let's use two separate error messages, one for each case, then make each error message clearer.